packages/router/utils.ts

```diff
@@ -1209,37 +1209,16 @@ export function resolveTo(
   // to the current location's pathname and *not* the route pathname.
   if (toPathname == null) {
     from = locationPathname;
-  } else if (isPathRelative) {
-    let fromSegments =
-      routePathnames.length === 0
-        ? []
-        : routePathnames[routePathnames.length - 1]
-            .replace(/^\//, "")
-            .split("/");
-
-    if (toPathname.startsWith("..")) {
-      let toSegments = toPathname.split("/");
-
-      // With relative="path", each leading .. segment means "go up one URL segment"
-      while (toSegments[0] === "..") {
-        toSegments.shift();
-        fromSegments.pop();
-      }
-
-      to.pathname = toSegments.join("/");
-    }
```

Comment:
This logic already happens in resolvePath below - we just need to pick the proper `from` path